tests/specs/geoQuery.spec.js

```diff
@@ -1376,5 +1376,29 @@ describe("GeoQuery Tests:", function() {
         cl.x("p5");
       }).catch(failTestOnCaughtError);
     });
+
+    it("Calling cancel() in the middle of firing \"key_entered\" events is allowed", function(done) {
+      var cl = new Checklist(["p1", "cancel query", "loc1 entered"], expect, done);
+
+      geoQueries.push(geoFire.query({center: [1,2], radius: 1000}));
+
+      geoFire.set({
+        "loc1": [1, 2],
+        "loc2": [1, 3],
+        "loc3": [1, 4]
+      }).then(function() {
+        cl.x("p1");
+
+        var numKeyEnteredEventsFired = 0;
+        geoQueries[0].on("key_entered", function(key, location, distance) {
+          cl.x(key + " entered");
+          numKeyEnteredEventsFired++;
+          if (numKeyEnteredEventsFired === 1) {
+            cl.x("cancel query");
+            geoQueries[0].cancel();
```

Comment:
Should this test also ensure that no more events are fired? if so (and I think it's probably the only sane behavior) we need to make sure we fail on that case, else it's probably fine as is.